### PR TITLE
Adding Prefix Checks to Folder Find Logic

### PR DESCRIPTION
**Pull request by luthermonson**
Fixing folders to work with deeper folder structures exposed a this old logic from the driver which prefixed folders when we are now passing in fully pathed folders. Leaving old system as a fall back for old nodetemplates.

https://github.com/rancher/rancher/issues/24840

### CODE REVIEW

**Comment by luthermonson** [comment]
> we're using same logic thrice, would it make sense to put it to a common func?

sure, good idea. i have a better way to do it too gimme a sec.

**Comment by dramich** [comment]
Use path.Join when building paths.

File: drivers/vmwarevsphere/utils.go [hunk 1]
```diff
@@ -265,3 +267,23 @@ func (d *Driver) restLogin(ctx context.Context, c *vim25.Client) (*library.Manag
...
 }
+
+func (d *Driver) findFolder() (*object.Folder, error) {
+	folders, err := d.datacenter.Folders(d.getCtx())
+	if err != nil {
+		return nil, err
+	}
+
+	// root vm folder default
+	folder := folders.VmFolder
```
Comment (dramich) [inline drivers/vmwarevsphere/utils.go]: This can be moved down to the return statement, no reason to initialize it here since it isn't used

Comment (luthermonson) [reply drivers/vmwarevsphere/utils.go]: sure

**Pull request merged**